docs/plugins/replace.md

```diff
@@ -0,0 +1,213 @@
+# Replace Plugin
+
+The `replacePlugin` is a built-in Rolldown plugin that replaces targeted strings in files during bundling. It's commonly used for injecting environment variables, build constants, and feature flags into your code.
+
+## Why This Is Needed
+
+When bundling JavaScript applications, you often need to replace placeholder strings with actual values at build time. Common scenarios include:
+
+- Converting `process.env.NODE_ENV` to `"production"` to enable optimizations and dead code elimination
+- Injecting API endpoints that differ between development and production
+- Embedding build metadata like version numbers and timestamps
+- Toggling feature flags without runtime overhead
+
+By performing these replacements during bundling, the final code contains literal values that can be optimized by minifiers and don't require runtime lookups.
+
+## Usage
+
+```js
+import { defineConfig } from 'rolldown';
+import { replacePlugin } from 'rolldown/plugins';
+
+export default defineConfig({
+  input: 'src/index.js',
+  output: {
+    dir: 'dist',
+    format: 'esm',
+  },
+  plugins: [
+    replacePlugin({
+      'process.env.NODE_ENV': JSON.stringify('production'),
+      '__DEV__': 'false',
+      '__VERSION__': JSON.stringify('1.0.0'),
+    }),
+  ],
+});
+```
+
+## Options
+
+### Basic Syntax
+
+```js
+replacePlugin(values, options);
+```
+
+- **`values`**: An object mapping strings to their replacements
+- **`options`**: Optional configuration for fine-tuning behavior
+
+### `values`
+
+Type: `Record<string, string | number>`
+
+An object where keys are the strings to search for and values are their replacements.
+
+```js
+replacePlugin({
+  'process.env.NODE_ENV': JSON.stringify('production'),
+  '__DEV__': 'false',
+  '__API_URL__': JSON.stringify('https://api.example.com'),
+});
+```
+
+**Important**: Use `JSON.stringify()` for string values to ensure they're properly quoted in the output.
```

Comment:
We should use a [GitHub-flavored alert](https://vitepress.dev/guide/markdown#github-flavored-alerts) here


```suggestion

> [!IMPORTANT]
> Use `JSON.stringify()` for string values to ensure they're properly quoted in the output.
```